kubert/src/admin.rs

```diff
@@ -132,23 +134,33 @@ impl Bound {
     /// Binds and runs the server on a background task, returning a handle
     pub fn spawn(self) -> Server {
         let ready = self.ready.clone();
-        let server = self
-            .server
-            .serve(hyper::service::make_service_fn(move |_conn| {
-                let ready = ready.clone();
-                future::ok::<_, hyper::Error>(hyper::service::service_fn(
-                    move |req: hyper::Request<hyper::Body>| match req.uri().path() {
-                        "/live" => future::ok(handle_live(req)),
-                        "/ready" => future::ok(handle_ready(&ready, req)),
-                        _ => future::ok::<_, hyper::Error>(
-                            Response::builder()
-                                .status(hyper::StatusCode::NOT_FOUND)
-                                .body(hyper::Body::default())
-                                .unwrap(),
-                        ),
-                    },
-                ))
-            }));
+        let metrics = PrometheusBuilder::new()
+            .install_recorder()
```

Comment:
I wonder if we want the `Builder` and/or `AdminArgs` types to expose some of [`PrometheusBuilder`'s other configuration options](https://docs.rs/metrics-exporter-prometheus/latest/metrics_exporter_prometheus/struct.PrometheusBuilder.html)?

In particular, it would probably be nice to expose the [`with_idle_timeout`](https://docs.rs/metrics-exporter-prometheus/latest/metrics_exporter_prometheus/struct.PrometheusBuilder.html#method.idle_timeout) and [`add_global_label`](https://docs.rs/metrics-exporter-prometheus/latest/metrics_exporter_prometheus/struct.PrometheusBuilder.html#method.add_global_label) configuration options. 

On the other hand, we definitely don't want to expose any of the options related to `metrics-exporter-prometheus`'s HTTP listener support, since we're serving metrics on the same port as the kubert admin server, rather than using the prometheus exporter's built-in HTTP endpoint. It might be nice to re-implement some of the same functionality (such as setting allowed addresses for the metrics endpoint) in follow-up PRs, though.

Reply:
I don't think we care about any of this in the short term.

It's worth nothing that we need to allow applications, however, to interact with the builder so that they can configure histograms, timeout, etc... So I don't think we can build the PrometheusBuilder here...